time/src/time.rs

```diff
@@ -29,21 +29,105 @@ pub(crate) enum Padding {
 /// (either positive or negative).
 ///
 /// When comparing two `Time`s, they are assumed to be in the same calendar date.
-#[derive(Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
+#[derive(Clone, Copy)]
+#[repr(C)]
 pub struct Time {
+    // The order of this struct's fields matter!
+    // Do not change them.
+
+    // Little endian version
+    #[cfg(target_endian = "little")]
+    #[allow(clippy::missing_docs_in_private_items)]
+    nanosecond: u32,
+    #[cfg(target_endian = "little")]
+    #[allow(clippy::missing_docs_in_private_items)]
+    second: u8,
+    #[cfg(target_endian = "little")]
+    #[allow(clippy::missing_docs_in_private_items)]
+    minute: u8,
+    #[cfg(target_endian = "little")]
+    #[allow(clippy::missing_docs_in_private_items)]
+    hour: u8,
+    #[cfg(target_endian = "little")]
+    #[allow(clippy::missing_docs_in_private_items)]
+    padding: Padding,
+
+    // Big endian version
+    #[cfg(target_endian = "big")]
+    #[allow(clippy::missing_docs_in_private_items)]
+    padding: Padding,
+    #[cfg(target_endian = "big")]
     #[allow(clippy::missing_docs_in_private_items)]
     hour: u8,
+    #[cfg(target_endian = "big")]
     #[allow(clippy::missing_docs_in_private_items)]
     minute: u8,
+    #[cfg(target_endian = "big")]
     #[allow(clippy::missing_docs_in_private_items)]
     second: u8,
+    #[cfg(target_endian = "big")]
     #[allow(clippy::missing_docs_in_private_items)]
     nanosecond: u32,
-    #[allow(clippy::missing_docs_in_private_items)]
-    padding: Padding,
+}
+
+impl core::hash::Hash for Time {
+    fn hash<H: core::hash::Hasher>(&self, state: &mut H) {
+        self.as_u64().hash(state)
+    }
+}
+
+impl PartialEq for Time {
+    fn eq(&self, other: &Self) -> bool {
+        self.as_u64().eq(&other.as_u64())
+    }
+}
+
+impl Eq for Time {}
```

Comment:
I believe this can be derived even if `PartialEq` is manually implemented.

Reply:
Done :heavy_check_mark: